src/test/java/org/broadinstitute/consent/http/service/dao/DacServiceDAOTest.java

```diff
@@ -0,0 +1,132 @@
+    User user = createUser();
+    daaDAO.createDaa(user.getUserId(), new Date().toInstant(), user.getUserId(), new Date().toInstant(), dacsInDatabase.get(0).getDacId());
+    daaDAO.createDaa(user.getUserId(), new Date().toInstant(), user.getUserId(), new Date().toInstant(), dacsInDatabase.get(0).getDacId());
+    Dac fullDac = dacDAO.findById(dacsInDatabase.get(0).getDacId());
+
+    try {
+      serviceDAO.deleteDacAndDaas(fullDac);
+    } catch (Exception e) {
+      fail("Delete should not fail");
+    }
+    List<Dac> dacListRemoved = dacDAO.findAll();
+    assertEquals(3, dacListRemoved.size());
+    List<DataAccessAgreement> daaListRemoved = daaDAO.findAll();
+    assertEquals(0, daaListRemoved.size());
+  }
+
+  @Test
+  void testDeleteDacWithDaasAndBroadDaa() {
+    getDacs();
+    Integer dacId = dacDAO.createDac(RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10),  new Date());
+    Integer broadDacId = dacDAO.createDac("Broad", RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10),  new Date());
+    User user = createUser();
+    Integer daaId = daaDAO.createDaa(user.getUserId(), new Date().toInstant(), user.getUserId(), new Date().toInstant(), dacId);
+    DataAccessAgreement daa = daaDAO.findById(daaId);
+    Integer daaId2 = daaDAO.createDaa(user.getUserId(), new Date().toInstant(), user.getUserId(), new Date().toInstant(), broadDacId);
+    daaDAO.createDacDaaRelation(dacId, daaId2);
+
+    Dac fullDac = dacDAO.findById(dacId);
+
+    try {
+      serviceDAO.deleteDacAndDaas(fullDac);
+    } catch (Exception e) {
+      fail("Delete should not fail");
+    }
+    List<Dac> dacListRemoved = dacDAO.findAll();
+    assertEquals(1, dacListRemoved.size());
+    List<DataAccessAgreement> daaListRemoved = daaDAO.findAll();
+    assertEquals(1, daaListRemoved.size());
+  }
+
+  @Test
+  void testDeleteDacWithBroadDaa() {
+    getDacs();
+    Integer dacId = dacDAO.createDac(RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10),  new Date());
+    Integer broadDacId = dacDAO.createDac("Broad", RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10),  new Date());
+    User user = createUser();
+    Integer daaId = daaDAO.createDaa(user.getUserId(), new Date().toInstant(), user.getUserId(), new Date().toInstant(), broadDacId);
+    daaDAO.createDacDaaRelation(dacId, daaId);
+    Dac fullDac = dacDAO.findById(dacId);
+
+    try {
+      serviceDAO.deleteDacAndDaas(fullDac);
+    } catch (Exception e) {
+      fail("Delete should not fail");
+    }
+    List<Dac> dacListRemoved = dacDAO.findAll();
+    assertEquals(1, dacListRemoved.size());
+    List<DataAccessAgreement> daaListRemoved = daaDAO.findAll();
+    assertEquals(1, daaListRemoved.size());
+  }
+
+  /**
+   * @return A list of 5 dacs
+   */
+  private List<Dac> getDacs() {
```

Comment:
minor nit, naming. In the method calls, it looks like we're getting pre-existing DACs. Maybe something like `createMockDACs` would be more informative.